src/item/Food.php

```diff
@@ -41,4 +42,8 @@ public function getAdditionalEffects() : array{
 	public function onConsume(Living $consumer) : void{
 
 	}
+
+	public function canStartUsingItem(Player $player) : bool{
+		return !$player->hasFiniteResources() || !$this->requiresHunger() || $player->getHungerManager()->isHungry();
```

Comment:
We don't currently support eating in creative. This condition should be consistent with this one: https://github.com/pmmp/PocketMine-MP/blob/c77829f4adee4d5fe61829823f4dbad545c100e9/src/entity/Human.php#L168